### PR TITLE
Merge 3.x

### PR DESCRIPTION
**Pull request by greg0ire**
Maintenance merge

### CODE REVIEW

File: Controller/PostController.php [hunk 1]
```diff
@@ -381,7 +381,7 @@ protected function getBlog()
     private function resolveRequest(Request $request = null)
     {
         if (null === $request) {
-            return $this->getRequest();
+            return $this->get('request_stack')->getCurrentRequest();
```
Comment (jordisala1991) [inline Controller/PostController.php]: Wow, there is no BC layer for this on 3.x or you did remove it when merging?

File: Tests/Entity/PostManagerTest.php [hunk 2]
```diff
@@ -69,11 +68,52 @@ public function assertPostEnabled($qb, $flag)
         $qb->expects($this->once())->method('setParameters')->with($this->equalTo(array('enabled' => $flag)));
     }
 
+    public function testFindOneByPermalinkSlug()
+    {
+        $permalink = $this->getMock('Sonata\NewsBundle\Permalink\PermalinkInterface');
```
Comment (jordisala1991) [inline Tests/Entity/PostManagerTest.php]: this should be createMock (not sure if this was like this on 3.x, and should be fixed there first)

File: Controller/PostController.php (hunk 1 above)
Comment (greg0ire) [inline Controller/PostController.php]: Looks like it is missing on 3.x

File: Tests/Entity/PostManagerTest.php (hunk 2 above)
Comment (greg0ire) [inline Tests/Entity/PostManagerTest.php]: Should be fixed on 3.x

File: .travis.yml [hunk 3]
```diff
@@ -12,6 +12,7 @@ language: php
 php:
   - '5.6'
```
Comment (jordisala1991) [inline .travis.yml]: this one should be removed

File: .travis.yml [hunk 4]
```diff
@@ -33,29 +34,29 @@ env:
 matrix:
   fast_finish: true
   include:
-    - php: '7.0'
+    - php: '7.1'
       env: TARGET=docs
-    - php: '7.0'
+    - php: '7.1'
       env: TARGET=lint
     - php: '5.6'
```
Comment (jordisala1991) [inline .travis.yml]: this one should be 7.0

**Comment by jordisala1991** [comment]
Not sure if we should add it here or is a dev-kit issue (or this bundle is not compatible with SF3.x but travis is missing builds with it: Symfony 3.1.x and 3.2.x

**Comment by greg0ire** [comment]
> Not sure if we should add it here or is a dev-kit issue (or this bundle is not compatible with SF3.x but travis is missing builds with it: Symfony 3.1.x and 3.2.x

This is a 3.x merge. Once dev-kit is able to deliver updates again b/c there are no more conflicts, I think it will add it itself.

**Comment by jordisala1991** [comment]
Travis is failing due to CoreBundle requirements. I think this csn be fixed by changing on composer.json to '3.2'

**Comment by greg0ire** [comment]
@sonata-project/contributors please review and merge

**Comment by jordisala1991** [comment]
Thanks @greg0ire

**Pull request merged**